pkg/registry/core/pod/storage/eviction.go

```diff
@@ -179,7 +180,7 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 			deleteOptions = deleteOptions.DeepCopy()
 			setPreconditionsResourceVersion(deleteOptions, &pod.ResourceVersion)
 		}
-		err = addConditionAndDeletePod(r, ctx, eviction.Name, rest.ValidateAllObjectFunc, deleteOptions)
+		err = addConditionAndDeletePod(r, ctx, pod, deleteOptions)
```

Comment:
revert the signature change here, we're not using the looked-up pod at all any more inside addConditionAndDeletePod

Reply:
up for a discussion in https://github.com/kubernetes/kubernetes/pull/116554#discussion_r1139283729

Reply:
reverted